play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -263,27 +263,30 @@ class UserOptionsPanel extends PSRoomPanel {
 			cmd += data.duration === "1 week" ? "/weekban " : "/ban ";
 			cmd += `${targetUser} ${data.reason ? ',' + data.reason : ''}`;
 		}
-		targetRoom?.send(cmd);
+		targetRoom.send(cmd);
 		this.close();
 	};
 
 	handleAddFriend = (ev: Event) => {
-		const { targetUser, targetRoom } = this.getTargets();
-		targetRoom?.send(`/friend add ${targetUser}`);
+		let { targetUser, targetRoom } = this.getTargets();
+		if (!targetRoom) targetRoom = PS.room.getParent()?.getParent()?.getParent() as ChatRoom;
```

Comment:
Shouldn't this simply to inside `getTargets` rather than being repeated 3 times?

Reply:
I'd also suggest `||=`